azuredevops/internal/service/servicehook/tfs_publisher.go

```diff
@@ -0,0 +1,965 @@
+			}
+			if val, exists := inputs[0].(map[string]interface{})["area_path"]; exists && val.(string) != "" {
+				eventConfig["areaPath"] = val.(string)
+			}
+			if val, exists := inputs[0].(map[string]interface{})["tag"]; exists && val.(string) != "" {
+				eventConfig["tag"] = val.(string)
+			}
+		}
+	}
+	if inputsList := d.Get("work_item_restored"); len(inputsList.([]interface{})) > 0 {
+		eventType = "work_item_restored"
+		if inputs, ok := inputsList.([]interface{}); ok && inputs[0] != nil {
+			if val, exists := inputs[0].(map[string]interface{})["work_item_type"]; exists && val.(string) != "" {
+				eventConfig["workItemType"] = val.(string)
+			}
+			if val, exists := inputs[0].(map[string]interface{})["area_path"]; exists && val.(string) != "" {
+				eventConfig["areaPath"] = val.(string)
+			}
+			if val, exists := inputs[0].(map[string]interface{})["tag"]; exists && val.(string) != "" {
+				eventConfig["tag"] = val.(string)
+			}
+		}
+	}
+	if inputsList := d.Get("work_item_commented"); len(inputsList.([]interface{})) > 0 {
+		eventType = "work_item_commented"
+		if inputs, ok := inputsList.([]interface{}); ok && inputs[0] != nil {
+			if val, exists := inputs[0].(map[string]interface{})["work_item_type"]; exists && val.(string) != "" {
+				eventConfig["workItemType"] = val.(string)
+			}
+			if val, exists := inputs[0].(map[string]interface{})["area_path"]; exists && val.(string) != "" {
+				eventConfig["areaPath"] = val.(string)
+			}
+			if val, exists := inputs[0].(map[string]interface{})["tag"]; exists && val.(string) != "" {
+				eventConfig["tag"] = val.(string)
+			}
+			if val, exists := inputs[0].(map[string]interface{})["comment_pattern"]; exists && val.(string) != "" {
+				eventConfig["commentPattern"] = val.(string)
+			}
+		}
+	}
+	if inputsList := d.Get("service_connection_created"); len(inputsList.([]interface{})) > 0 {
+		eventType = "service_connection_created"
+		if inputs, ok := inputsList.([]interface{}); ok && inputs[0] != nil {
+			if val, exists := inputs[0].(map[string]interface{})["project"]; exists && val.(string) != "" {
+				eventConfig["project"] = val.(string)
+			}
+		}
+	}
+	if inputsList := d.Get("service_connection_updated"); len(inputsList.([]interface{})) > 0 {
+		eventType = "service_connection_updated"
+		if inputs, ok := inputsList.([]interface{}); ok && inputs[0] != nil {
+			if val, exists := inputs[0].(map[string]interface{})["project"]; exists && val.(string) != "" {
+				eventConfig["project"] = val.(string)
+			}
+		}
+	}
+
+	eventConfig["projectId"] = d.Get("project_id").(string)
+	return eventConfig, tfsResourceBlock2ApiType[eventType]
+}
+
+// flattenTfsEventConfig flattens the TFS event configuration to schema
+func flattenTfsEventConfig(subscription *servicehooks.Subscription) (string, []interface{}) {
+	eventType := tfsApiType2ResourceBlock[*subscription.EventType]
+	if isNilTfsEventConfig(*subscription.PublisherInputs) {
+		return eventType, []interface{}{nil}
```

Comment:
What is the motivation here to return a `[]any{nil}`, instead of skipping this check and return `[]any{map[string]any{}}`, since each case will end up with a plan diff?